autorest/sender.go

```diff
@@ -218,8 +218,7 @@ func DoRetryForStatusCodes(attempts int, backoff time.Duration, codes ...int) Se
 		return SenderFunc(func(r *http.Request) (resp *http.Response, err error) {
 			rr := NewRetriableRequest(r)
 			// Increment to add the first call (attempts denotes number of retries)
-			attempts++
-			for attempt := 0; attempt < attempts; {
+			for attempt := 0; attempt < attempts+1; {
```

Comment:
so we dont need to increment attempts here ?

Reply:
or is enough to increment it later ?

Reply:
It's done this way so that the first attempt isn't counted as a "retry attempt", i.e. if attempts is 5 it means 5 retry attempts which makes a total of six requests.